skfda/tests/test_grid.py

```diff
@@ -337,6 +337,19 @@ def test_evaluate_grid_unaligned(self) -> None:
 
         np.testing.assert_allclose(res, expected)
 
+    def test_restrict(self) -> None:
+        """ Test FDataGrid.restrict with bounds. """
+        # Test 1 sample function R^3 -> R^5.
+        grid_points = ([0, 1], [0, 1, 2], [0, 1, 2, 3])
+        data_matrix = np.ones((1, 2, 3, 4, 5))
+        fd = FDataGrid(data_matrix, grid_points)
+        restricted_domain = ((0, 1), (.5, 1.5), (.5, 2))
+        fd_restricted = fd.restrict(restricted_domain, with_bounds=True)
+        res = fd_restricted.grid_points
+        expected = ([0, 1], [.5, 1, 1.5], [.5, 1, 2])
```

Comment:
We should probably aid Mypy by explicitly typing this variable or using arrays instead of lists, as the type inferred for the tuple elements is `object`.

Reply:
Which of the following two (if any) would you prefer?
```python
expected = map(np.array, ([0, 1], [0.5, 1, 1.5], [0.5, 1, 2]))
expected: tuple[list] = ([0, 1], [0.5, 1, 1.5], [0.5, 1, 2])
```

Reply:
Second question then since it's too long:
```python
expected: Tuple[Sequence[float], ...] = ([0, 1],
                                         [0.5, 1, 1.5],
                                         [0.5, 1, 2])
# or
expected: Tuple[Sequence[float], ...] = (
    [0, 1],
    [0.5, 1, 1.5],
    [0.5, 1, 2],
)

```

Reply:
The second one is the one accepted by [our style checker of choice](https://github.com/wemake-services/wemake-python-styleguide).